zio-cache/shared/src/main/scala/zio/cache/ScopedCache.scala

```diff
@@ -108,220 +108,32 @@ object ScopedCache {
-          scopedEffect <- (for {
-                            scope <- Scope.make
-                            exit <- scopedLookup(key)
-                                      .provideEnvironment(environment.add[Scope](scope))
-                                      .exit
-                          } yield (exit, scope.close(_)))
-                            .onInterrupt(ZIO.succeed(map.remove(key)))
-                            .flatMap { case (exit, release) =>
-                              val now       = Instant.now(clock)
-                              val expiredAt = now.plus(timeToLive(exit))
-                              exit match {
-                                case Exit.Success(value) =>
-                                  val exitWithReleaser: Exit[Nothing, (Value, Finalizer)] =
-                                    Exit.succeed(value -> release)
-                                  val completedResult = MapValue
-                                    .Complete(
-                                      key = new MapKey(key),
-                                      exit = exitWithReleaser,
-                                      ownerCount = new AtomicInteger(1),
-                                      entryStats = EntryStats(now),
-                                      timeToLive = expiredAt
-                                    )
-                                  val previousValue = map.put(key, completedResult)
-                                  ZIO.succeed(
-                                    cleanMapValue(previousValue).as(completedResult.toScoped).flatten
-                                  )
-                                case failure @ Exit.Failure(_) =>
-                                  val completedResult =
-                                    MapValue.Complete(
-                                      key = new MapKey(key),
-                                      exit = failure,
-                                      ownerCount = new AtomicInteger(0),
-                                      entryStats = EntryStats(now),
-                                      timeToLive = expiredAt
-                                    )
-                                  val previousValue = map.put(key, completedResult)
-                                  release(failure) *> ZIO.succeed(
-                                    cleanMapValue(previousValue).as(completedResult.toScoped).flatten
-                                  )
-                              }
-                            }
-                            .memoize
-        } yield scopedEffect.flatten
-
-        private def hasExpired(timeToLive: Instant) =
-          Instant.now(clock).isAfter(timeToLive)
+      .flatMap { scopedCache =>
+        val cleaningInterval = 1.second
+        (scopedCache.freeExpired *> ZIO.sleep(cleaningInterval))
+          .repeat(Schedule.forever)
+          .interruptible
+          .fork
```

Comment:
Could just use `forkScoped` here and tie the life of the fiber to the life of the cache.

Reply:
I just pushed a commit about that. Tell me if this is done correctly.